### DISCUSSION TITLE
simplepyble

### DISCUSSION THREAD

**Discussion by baseplate-admin** [flet-dev/flet#3774] [Packages] [answered]
- [Github](https://github.com/OpenBluetoothToolbox/SimpleBLE)
- [PYPI](https://pypi.org/project/simplepyble/)

Written in:
- C++

**Comment by markusprograms** [comment]
So it's not possible to build an APK? Should I use another library? Which one?

**Reply by baseplate-admin** [reply]
I think there's no pure python alternative for this

**Comment by ndonkoHenri** [comment, marked as answer]
The `simplepyble` C++ extension itself would compile for mobile, but seemingly, it can't actually function on Android inside a Flet app.

- On Android, SimpleBLE's backend is a JNI bridge that, at startup, needs a live JVM/ART VM, the Android Context, and the `android.bluetooth.*` Java classes resolved via JNI (plus the runtime `BLUETOOTH_SCAN`/`CONNECT` permissions). 
- SimpleBLE is licensed BUSL-1.1 (commercial use requires a paid license), which is a separate distribution problem.

A possible alternative is via the native-API bridges: [pyjnius/pyobjus](https://flet.dev/blog/tap-into-native-android-and-ios-apis-with-Pyjnius-and-pyobjus/). In the future we could work on supporting [`flutter_blue_plus`](https://pub.dev/packages/flutter_blue_plus) (https://github.com/flet-dev/flet/discussions/5181).
